tests/unit_tests/commandline/cli_top_level_tester.py

```diff
@@ -125,33 +262,139 @@ def check_output(self, args, output):
+            except SystemExit as se:
+                print(se, file=sys.stderr)
+            except Exception as e:
+                print(e, file=sys.stderr)
+            # except RuntimeError as re:
+            #     print(re, file=sys.stderr)
+
+        # Retrieve the output printed from 'func'
+        stdout = stdout_capture.getvalue()
+        stderr = stderr_capture.getvalue()
+
+        # Close the StringIO Objects
+        stdout_capture.close()
+        stderr_capture.close()
 
-        This test covers every valid combination of commands for the
-        'geoips list <sub-cmd> ...' command. We also test invalid commands, to ensure
-        that the proper help documentation is provided for those using the command
-        incorrectly.
+        return stdout, stderr
+
+    def viable_monkeypatch(self, args):
+        """Determine whether or not the arguments are viable to run via monkeypatch.
+
+        We are currently unable to catch certain commands outputs using
+        contextlib.redirect_stdout / contextlib.redirect_stderr. The cause of this is
+        still unknown, however we have a workaround using subprocess for the time being.
+
+        If the arguments can be ran via monkeypatch, return True, otherwise return
+        False.
 
         Parameters
         ----------
-        args: 2D array of str
-            - List of arguments to call the CLI with (ie. ['geoips', 'list <sub-cmd>'])
+        args: array of str
+            - List of arguments to call the CLI with (ie. ['geoips', '<cmd>'])
+
+        Returns
+        -------
+        monkeypatch_viable: bool
+            - Whether or not the arguments can be ran with monkeypatch. Monkeypatch is a
+              much quicker option if available.
+        """
+        match args:
+            case _ if "-h" in args:
+                # Can't capture help messages using monkeypatch... yet
+                return False
+            case _ if "linting" in args:
+                # Can't capture linting output using monkeypatch... yet
+                return False
+            case _ if ("test" in args and "script" in args):
+                # Can't capture bash script output using monkeypatch... yet
+                return False
+            case _ if (
+                "run" in args
+                or "run_procflow" in args
+                or "data_fusion_procflow" in args
+            ):
+                # Can't capture procflow output using monkeypatch... yet
+                return False
+            case _ if any(["non_existent" in arg for arg in args]):
+                # Can't capture argparse.ArgumentError output using monkeypatch... yet
+                return False
+            case _ if ("--long" in args and "--columns" in args):
+                # Can't capture argparse.ArgumentError output using monkeypatch... yet
+                return False
+            case _:
+                # Monkeypatch works for the provided arguments!
+                return True
```

Comment:
I think that a lot of this could be fixed by using pytest's built-in methods of capturing stdout/stderr. See here: https://docs.pytest.org/en/7.1.x/how-to/capture-stdout-stderr.html

Reply:
I'll write an issue to explore ``pytest.capsys`` in another PR and create an issue for this.

Reply:
See #628.